src/components/htmlvideoplayer/plugin.js

```diff
@@ -942,129 +1004,141 @@ define(['browser', 'require', 'events', 'apphost', 'loading', 'dom', 'playbackMa
             });
         }
 
-        function fetchSubtitles(track, item) {
+        /**
+         * @private
+         */
+        fetchSubtitles(track, item) {
             if (window.Windows && itemHelper.isLocalItem(item)) {
-                return fetchSubtitlesUwp(track, item);
+                return this.fetchSubtitlesUwp(track, item);
             }
 
-            incrementFetchQueue();
-            return new Promise(function (resolve, reject) {
-                var xhr = new XMLHttpRequest();
+            this.incrementFetchQueue();
+            return new Promise((resolve, reject) => {
+                const xhr = new XMLHttpRequest();
 
-                var url = getTextTrackUrl(track, item, '.js');
+                const url = getTextTrackUrl(track, item, ".js");
 
                 xhr.open('GET', url, true);
 
-                xhr.onload = function (e) {
+                xhr.onload = (e) => {
                     resolve(JSON.parse(this.response));
-                    decrementFetchQueue();
+                    this.decrementFetchQueue();
                 };
 
-                xhr.onerror = function (e) {
+                xhr.onerror = (e) => {
                     reject(e);
-                    decrementFetchQueue();
+                    this.decrementFetchQueue();
                 };
 
                 xhr.send();
             });
         }
 
-        function setTrackForDisplay(videoElement, track) {
-
+        /**
+         * @private
+         */
+        setTrackForDisplay(videoElement, track) {
             if (!track) {
-                destroyCustomTrack(videoElement);
+                this.destroyCustomTrack(videoElement);
                 return;
             }
 
             // skip if already playing this track
-            if (customTrackIndex === track.Index) {
+            if (this.customTrackIndex === track.Index) {
                 return;
             }
 
-            self.resetSubtitleOffset();
-            var item = self._currentPlayOptions.item;
+            this.resetSubtitleOffset();
+            const item = self._currentPlayOptions.item;
 
-            destroyCustomTrack(videoElement);
-            customTrackIndex = track.Index;
-            renderTracksEvents(videoElement, track, item);
-            lastCustomTrackMs = 0;
+            this.destroyCustomTrack(videoElement);
+            this.customTrackIndex = track.Index;
+            this.renderTracksEvents(videoElement, track, item);
+            this.lastCustomTrackMs = 0;
         }
 
-        function renderSsaAss(videoElement, track, item) {
-            var attachments = self._currentPlayOptions.mediaSource.MediaAttachments || [];
-            var apiClient = connectionManager.getApiClient(item);
-            var options = {
+        /**
+         * @private
+         */
+        renderSsaAss(videoElement, track, item) {
+            const attachments = self._currentPlayOptions.mediaSource.MediaAttachments || [];
```

Comment:
Shouldn't `self` be `this` here? (To reference the class instance instead of `window.self`).

Reply:
Totally missed that one